src/fw/drivers/imu/lsm6dso/lsm6dso.c

```diff
@@ -432,6 +437,63 @@ static void prv_lsm6dso_process_interrupts(void) {
             axis_offset, axis_direction);
     accel_cb_double_tap_detected(axis_offset, axis_direction);
   }
+
+  // Wake-up (any-motion) event -> treat as shake. Axis & direction derived from wake_up_src.
+  if (s_lsm6dso_state.shake_detection_enabled && all_sources.wake_up) {
+    lsm6dso_wake_up_src_t wake_src;
+    if (lsm6dso_read_reg(&lsm6dso_ctx, LSM6DSO_WAKE_UP_SRC, (uint8_t *)&wake_src, 1) == 0) {
+      IMUCoordinateAxis axis = AXIS_X;
+      int32_t direction = 1; // LSM6DSO does not give sign directly for wake-up; approximate via sign of latest sample on axis
+      // Determine which axis triggered: order X,Y,Z
+      const AccelConfig *cfg = &BOARD_CONFIG_ACCEL.accel_config;
+      if (wake_src.x_wu) {
+        axis = AXIS_X;
+      } else if (wake_src.y_wu) {
+        axis = AXIS_Y;
+      } else if (wake_src.z_wu) {
+        axis = AXIS_Z;
+      }
+      // Read current sample to infer direction
+      int16_t accel_raw[3];
```

Comment:
Is this sufficient? The direction will almost always be biased in the direction of gravity, which not necessarily the shake direction. If you take this approach, you'll likely need a two-sample delta at least?